engines/scc_proxy/lib/scc_proxy/engine.rb

```diff
@@ -172,7 +172,7 @@ def scc_check_subscription_expiration(headers, login, system_token, logger)
       active_products_ids = scc_systems_activations.map { |act| act['service']['product']['id'] if act['status'].casecmp('active').zero? }.flatten
       products = Product.where(id: active_products_ids)
       product_paths = products.map { |prod| prod.repositories.pluck(:local_path) }.flatten
-      active_subscription = product_paths.any? { |path| headers['X-Original-URI'].include?(path) }
+      active_subscription = product_paths.any? { |path| headers['X-Original-URI'].to_s.include?(path) }
```

Comment:
The more beautiful way to solve this would be: 
```suggestion
      active_subscription = product_paths.any? { |path| headers.fetch('X-Original-URI', '').include?(path) }
```

Reply:
Will update the code with the suggestion

Does it mean we should adapt that kind of checks with the suggestion ?
like [here](https://github.com/SUSE/rmt/blob/fc9f05171157bb0fa06282812ee3870baa9acacd/engines/zypper_auth/lib/zypper_auth/engine.rb#L10) ?
I followed the same style/pattern, happy to update it in the rest of the code 
I understand the code I shared is different and works as it is, I ask as a matter of same style
```ruby
# change
instance_data = Base64.decode64(request.headers['X-Instance-Data'].to_s)
# to
instance_data = Base64.decode64(request.headers.fetch('X-Instance-Data', ''))
```

Reply:
It's better style to define the default value explicitly and not rely on `nil.to_s`. But imho no need to change the whole codebase, just saw that in your new code.